foreign_cc/private/make_env_vars.bzl

```
@@ -30,18 +31,33 @@ def get_make_env_vars(
     return " ".join(["{}=\"{}\""
         .format(key, _join_flags_list(workspace_name, vars[key])) for key in vars])
 
-def _define_deps_flags(deps, inputs):
+def _define_deps_flags(deps, inputs, linkdeps):
     # It is very important to keep the order for the linker => put them into list
     lib_dirs = []
+    libs = []
 
     # Here go libraries built with Bazel
     gen_dirs_set = {}
+    lib_set = {}
     for lib in inputs.libs:
         dir_ = lib.dirname
         if not gen_dirs_set.get(dir_):
             gen_dirs_set[dir_] = 1
             lib_dirs.append("-L$$EXT_BUILD_ROOT$$/" + dir_)
 
+    if linkdeps:
+        for lib in reversed(inputs.libs):
+            libname = lib.basename
+            ext = lib.extension
+            if ext:
+                libname = libname[:-1-len(ext)]
+            if libname.startswith('lib'):
+                libname = libname[3:]
+            libname = "-l" + libname
+            if not lib_set.get(libname):
+                lib_set[libname] = 1
```

I am guessing this is some starlark limitation that we can't write this in a more pythonic way? Like for instance
```suggestion
            if not libname in lib_set:
                lib_set.add(libname)
```

---

Apparently starlark doesn't support `set`s. https://docs.bazel.build/versions/main/skylark/language.html#differences-with-python . I kept the same style as `gen_dirs_set` in this function